syntax = "proto3";

package aea.dvilela.kv_storage.v0_1_0;

message KvStorageMessage{

  // Performatives and contents
  message Read_Request_Performative{
    repeated string keys = 1;
  }

  message Read_Response_Performative{
    map<string, string> data = 1;
  }

  message Create_Or_Update_Request_Performative{
    map<string, string> data = 1;
  }

  message Success_Performative{
    string message = 1;
  }

  message Error_Performative{
    string message = 1;
  }


  oneof performative{
    Create_Or_Update_Request_Performative create_or_update_request = 5;
    Error_Performative error = 6;
    Read_Request_Performative read_request = 7;
    Read_Response_Performative read_response = 8;
    Success_Performative success = 9;
  }
}
